### ISSUE TITLE
Low quality window plots on hidpi display

### ISSUE THREAD

**Issue by chesnokov-yuriy** [matplotlib/matplotlib#17440]
### Bug report

**Bug summary**

Executing python script with line plotting code from command line on a hidpi 4k resolution display produces low quality UI window plot.

**Code for reproduction**

```bash
$ python3 plot.py
```

plot.py file contents:

```python
from numpy import random

import matplotlib as mpl
import matplotlib.pyplot as pp

print(mpl.__version__)
print(mpl.get_backend())

pp.figure()
pp.plot(random.randn(250))
pp.grid()
pp.show()
```

**Actual outcome**

![plot-lowdpi](https://user-images.githubusercontent.com/2661785/82206351-bc8b6580-9910-11ea-82f7-b2af07cdbefc.png)

```
3.2.1
Qt5Agg
```

**Expected outcome**

Expected high quality UI window plots as in Spyder IDE running on the same machine

![plot-hidpi](https://user-images.githubusercontent.com/2661785/82206342-ba290b80-9910-11ea-9666-6c2a0894702e.png)

or similar high quality UI window plots from command line in the previous Kubuntu 19 with matplotlib 3.0.3 and Python 3.7

**Matplotlib version**

  * Operating system: Kubuntu 20.04
  * Matplotlib version: 3.2.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.8.2
  * Jupyter version (if applicable):
  * Other libraries:

**Comment by timhoffm** [comment]
This is likely addressed by #15656.

What is the display scaling factor you are using? (probably something between 1 and 2)?

**Comment by chesnokov-yuriy** [comment]
![screen](https://user-images.githubusercontent.com/2661785/82263215-13be2400-996b-11ea-80a8-b4ab983f5920.png)

**Comment by tacaswell** [comment]
@chesnokov-yuriy If you change your scaling does it get better?  I agree with @timhoffm this is likely due to Qt growing the ability to do fractional scaling under us.

**Comment by chesnokov-yuriy** [comment]
@tacaswell let me try any different scaling and investigate #15656

```
Kubuntu 19 + matplotlib 3.0.3 command line plot.py: high quality
Kubuntu 19 + matplotlib 3.0.3 Spyder IDE plot: high quality
Kubuntu 19 + matplotlib 3.0.3 Jupyter notebook plot: high quality
```
on Kubuntu 20.04 situation became different somehow

```
Kubuntu 19 + matplotlib 3.0.3 command line plot.py: low quality
Kubuntu 19 + matplotlib 3.2.1 command line plot.py: low quality
Kubuntu 19 + matplotlib 3.2.1 Spyder IDE plot: high quality
Kubuntu 19 + matplotlib 3.2.1 Jupyter notebook plot: high quality only after %config InlineBackend.figure_format = 'retina'
Kubuntu 19 + matplotlib 3.2.1 IPython plot: low quality even after %config InlineBackend.figure_format = 'retina'
```

Only Spyder IDE produces high quality plots both inline and in QtAgg5 UI window even without any 'retina' settings

**Comment by timhoffm** [comment]
@chesnokov-yuriy do you know which Qt version each of the call uses?

Remarks:
- The effect is definitively dependent on the backend. In the original post, you should a window from the Qt backend.
- The inline backend in jupyter notebook is another piece of cake (though it might as well struggle with fractional magnifications).
- The InlineBackend config does not have an effect for IPython, because IPython cannot use the inline backend. It will likely also use the Qt backend.
- For spyder: do you pop up the plot in a separate window or is the plot integrated in the 
 Spyder IDE window?

**Comment by chesnokov-yuriy** [comment]
@timhoffm I do not remember which Qt was in Kubuntu 19.10 or 19.04 and which Plasma if only looked up from release specifications. Kubuntu 20.05 Qt is 5.12.8 Kde Plasma 5.18.4

Spyder 4.1.3 provides high quality plots in either inline or Qt5Agg without any additional settings
![spyder](https://user-images.githubusercontent.com/2661785/82334358-89200800-99f0-11ea-99b7-c3948c270815.png)
![preferences](https://user-images.githubusercontent.com/2661785/82335196-abfeec00-99f1-11ea-9ea8-35272cd8a9c9.png)

**Comment by chesnokov-yuriy** [comment]
as IPython QtConsole is in low quality
![ipython](https://user-images.githubusercontent.com/2661785/82335392-eb2d3d00-99f1-11ea-9e02-29bfea478d51.png)

**Comment by chesnokov-yuriy** [comment]
> @chesnokov-yuriy If you change your scaling does it get better?

display configuration settings global scale 100%: high quality plot

![scale1 0](https://user-images.githubusercontent.com/2661785/82386442-1721e000-9a3d-11ea-85bc-9d98dd9b248b.png)

display configuration settings global scale 125%: low quality same as 150% sa reported before

**Comment by chesnokov-yuriy** [comment]
> @chesnokov-yuriy do you know which Qt version each of the call uses?

Kubuntu 19.10 Qt 5.12.4 Python 3.7
Kubuntu 20.04 Qt 5.12.8 Python 3.8

**Comment by tacaswell** [comment]
> display configuration settings global scale 100%: high quality plot

That tracks really well with @timhoffm 's theory that this is related to us not handling the fractional scaling correctly.

**Comment by chesnokov-yuriy** [comment]
qapp.py file

```python
from matplotlib.backends.qt_compat import QtWidgets

qApp = QtWidgets.QApplication([])

print('  logical dpi:', qApp.desktop().logicalDpiX())
print(' physical dpi:', qApp.desktop().physicalDpiX())
print('        width:', qApp.desktop().screenGeometry().width())
print('       height:', qApp.desktop().screenGeometry().height())
```

bash shell environment

```
$ printenv | grep QT_
QT_ACCESSIBILITY=1
QT_SCREEN_SCALE_FACTORS=HDMI-1=1.5;HDMI-2=1.5;DP-1=1.5;
QT_AUTO_SCREEN_SCALE_FACTOR=0
```

display configuration: 150% scale with 3840x2160

```
$ python3 qapp.py
  logical dpi: 96
 physical dpi: 108
        width: 2560
       height: 1440
```

adding QT_SCALE_FACTOR environment variable set to 0.666667 which is 100/150%

```
$ QT_SCALE_FACTOR=0.666667 python3 qapp.py
  logical dpi: 96
 physical dpi: 163
        width: 3840
       height: 2160
```

using the same QT_SCALE_FACTOR set before plot.py call results in high quality plot with tiny toolbar and status bar, which looks much better though than default large margins

```
$ QT_SCALE_FACTOR=0.666667 python3 plot.py
```

![plot](https://user-images.githubusercontent.com/2661785/82478916-4cc8d680-9ada-11ea-9573-f3ed85a43e64.png)

**Comment by QuLogic** [comment]
Is this fixed by #15656?

**Comment by tacaswell** [comment]
Yes, it should be.
